File: Scripts/newTask/palindrome-detection/swift/palindrome-detection.swift
extension String {
    
    /**
     Checks if string is a palindrome, ignoring spaces and capitalization. [Source on GitHub](http://bit.ly/SwiftStringPalindromeExtension)
     - note:
     [What's a palindrome?](https://en.wikipedia.org/wiki/Palindrome)
     - returns: Bool
     */
    public func isPalindrome() -> Bool {
        var s = String(self.characters.filter { $0 != Character(" ") }).lowercased()
        return s == String(s.characters.reversed())
    }
    
}

for i in 0...100000000 {
 let data = "saippuakivikauppias"
 data.isPalindrome()   // true
}

File: Scripts/newTask/palindrome-detection/go/palindrome-detection.go
package main

func main() {

for i := 0; i < 1000000000; i++ {
	var flag bool
	flag = isPalindrome("saippuakivikauppias")
	_ = flag
}
}

func isPalindrome(s string) bool {
	runes := []rune(s)
	for len(runes) > 1 {
		if runes[0] != runes[len(runes)-1] {
			return false
		}
		runes = runes[1 : len(runes)-1]
	}
	return true
}
